src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,206 @@
+
+    fields: list[FieldDict]
+    metadata: list[MetadataDict]
+    required: list[str]
+
+
+github_app_inputs: GitHubAppInputs = {
+    'fields': [
+        {
+            'id': 'github_api_url',
+            'label': _('GitHub API endpoint URL'),
+            'type': 'string',
+            'help_text': _(
+                'Specify the GitHub API URL here. In the case of an Enterprise: '
+                ' https://gh.your.org/api/v3 (self-hosted) '
+                'or https://api.SUBDOMAIN.ghe.com (cloud)',
+            ),
+            'default': 'https://api.github.com',
+        },
+        {
+            'id': 'app_id',
+            'label': _('GitHub App ID'),
+            'type': 'string',
+            'help_text': _(
+                'The GitHub App ID created by the GitHub Admin. '
+                'Example App ID: 1121547 '
+                'found on https://github.com/settings/apps/ '
+                'required for creating a JWT token for authentication',
+            ),
+        },
+        {
+            'id': 'install_id',
+            'label': _('GitHub App Installation ID'),
+            'type': 'string',
+            'help_text': _(
+                'The Installation ID from the GitHub App installation '
+                'generated by the GitHub Admin. '
+                'Example: 59980338 extracted from the installation link: '
+                'https://github.com/settings/installations/59980338 '
+                'required for creating a limited GitHub app token',
+            ),
+        },
+        {
+            'id': 'private_rsa_key',
+            'label': _('RSA Private Key'),
+            'type': 'string',
+            # `ssh_private_key` validator checks for PKCS#1 PEM-like formats
+            'format': 'ssh_private_key',
+            'secret': True,
+            'multiline': True,
+            'help_text': _(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation IDs.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'description',
+            'label': _('Description (Optional)'),
+            'type': 'string',
+            'help_text': _('To be removed after UI is updated'),
+        },
+    ],
+    'required': ['app_id', 'install_id', 'private_rsa_key'],
+}
+
+
+class EmptyKwargs(TypedDict):
+    """Schema for zero keyword arguments."""
+
+
+class MaybeBaseURLKwarg(TypedDict, total=False):
+    """Schema for optional PyGitHub ``base_url`` keyword argument."""
+
+    base_url: str
+
+
+def _validate_inputs(  # noqa: DAR101
+    app_id: str, install_id: str,
+) -> None:
+    if not app_id.isdigit():
+        raise ValueError(
+            f'Expected GitHub App ID to be an integer but got {app_id !r}',
+        )
+    if not install_id.isdigit():
+        raise ValueError(
+            f'Expected GitHub App ID to be an integer but got {app_id !r}',
```

Comment:
This string got broken. It's a copy of the App ID error but before this commit, it was rendering the installation id and had the word Installation in the middle. The change broke another test that still expect correct string here. I'll correct this as well.